client/pages/search.jsx

```diff
@@ -5,41 +5,43 @@ export default class Search extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      longitude: null,
-      latitude: null,
+      geoLocationOn: this.props.geoLocation,
+      location: { city: this.props.city, stateCode: this.props.stateCode },
       restaurants: [],
       isLoading: true
     };
   }
 
   componentDidMount() {
-    navigator.geolocation.getCurrentPosition(pos => {
-      const { longitude, latitude } = pos.coords;
-      const body = {
-        attribute: ['reservation'],
-        longitude,
-        latitude,
-        limit: 10
-      };
-      const searchRequest = {
-        method: 'PUT',
-        headers: {
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify(body)
-      };
-      fetch('api/yelp-search', searchRequest)
-        .then(response => response.json())
-        .then(data => {
-          this.setState({ restaurants: data.jsonBody.businesses, isLoading: false });
-        })
-        .catch(err => console.error(err));
-    });
+    if (this.state.geoLocationOn) {
+      navigator.geolocation.getCurrentPosition(pos => {
+        const { longitude, latitude } = pos.coords;
+        const body = {
+          attribute: ['reservation'],
+          longitude,
+          latitude,
+          limit: 10
+        };
+        const searchRequest = {
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(body)
+        };
+        fetch('/api/yelp-search/geolocation', searchRequest)
+          .then(response => response.json())
+          .then(data => {
+            this.setState({ restaurants: data.jsonBody.businesses, isLoading: false });
+          })
+          .catch(err => console.error(err));
+      });
+    }
   }
 
   render() {
     return this.state.isLoading
-      ? <p>loading..</p>
+      ? <img style={{ width: '200px', height: '200px' }}src="https://www.superiorlawncareusa.com/wp-content/uploads/2020/05/loading-gif-png-5.gif"></img>
```

Comment:
Please download the gif and save it to your `server/public/images` directory instead of referring to an external URL. 

If they take it down, your loading image will break.